### PR TITLE
Clarify overwrite behavior on upload methods

### PR DESCRIPTION
**Pull request by jaschrep-msft**
Rephrased the summary for upload methods on BlobClient and BlockBlobClient.

Some overloads mislead users regarding overwrite behavior. These updates are intended to fix the descriptions of those overloads, as well as generally advise users of all overloads what the default behavior of that method is and how to change that behavior. The focus is on alerting users as soon as possible whether a particular upload overwrites by default.

Goal of review is to determine the best language to use for this messaging, as well as verify these descriptions are on the correct overloads.

### CODE REVIEW

**Comment by jaschrep-msft** [comment]
@mhopkins-msft please advise on language for this change.

**Comment by mhopkins-msft** [comment]
Can we be any more specific about what exception is thrown so devs know what to catch?

**Comment by jaschrep-msft** [comment]
@mhopkins-msft, the remarks section of each of these methods states `A <see cref="RequestFailedException"/> will be thrown if a failure occurs.`. Is this not enough?

File: sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs [hunk 1]
```diff
@@ -756,10 +747,14 @@ protected internal virtual BlobClient WithClientSideEncryptionOptionsCore(Client
 
         /// <summary>
         /// The <see cref="Upload(Stream, BlobHttpHeaders, Metadata, BlobRequestConditions, IProgress{long}, AccessTier?, StorageTransferOptions, CancellationToken)"/>
-        /// operation creates a new block blob or updates the content of an
-        /// existing block blob.  Updating an existing block blob overwrites
+        /// operation overwrites the contents of the blob, creating a new block
+        /// blob if none exists.  Overwriting an existing block blob overwrites
```
Comment (mhopkins-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobClient.cs:751]: We're using "overwrite" a lot. Can we say: 

"Overwriting an existing block blob replaces any existing metadata on the blob."

**Review by mhopkins-msft — APPROVED** [review]

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**